### PR TITLE
React icons documentation revamp

### PR DESCRIPTION
**Pull request by tomi-msft**
This PR adds more documentation to the Icon page, including new icon packages, usage, and pros and cons 

Part of #23420

### CODE REVIEW

**Comment by fabricteam** [comment]
## 📊 Bundle size report

<details>
<summary>Unchanged fixtures</summary>

| Package & Exports                                                                                                                                                             |          Size (minified/GZIP) |
| ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ----------------------------: |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/MultipleComponents.fixture.js'>react-components: Accordion, Button, FluentProvider, Image, Menu, Popover</abbr> | `184.781 kB`<br />`51.176 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ProviderAndTheme.fixture.js'>react-components: FluentProvider & webLightTheme</abbr>                            |  `33.988 kB`<br />`11.108 kB` |

</details>
<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/bd64779b7d68775260b587f10265d64b916f6140'>bd64779b7d68775260b587f10265d64b916f6140</a></sub>
<div id="bundleSizeComment9423"/>

**Comment by fabricteam** [comment]
## [Perf Analysis (`@fluentui/react-components`)](https://github.com/microsoft/fluentui/wiki/Perf-Testing)
<p>No significant results to display.</p><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>Avatar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23438/perf-test-react-components/Avatar-mount_base.html">830</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23438/perf-test-react-components/Avatar-mount.html">895</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Button</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23438/perf-test-react-components/Button-mount_base.html">507</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23438/perf-test-react-components/Button-mount.html">558</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProvider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23438/perf-test-react-components/FluentProvider-mount_base.html">1807</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23438/perf-test-react-components/FluentProvider-mount.html">1790</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23438/perf-test-react-components/FluentProviderWithTheme-mount_base.html">268</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23438/perf-test-react-components/FluentProviderWithTheme-mount.html">263</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23438/perf-test-react-components/FluentProviderWithTheme-virtual-rerender_base.html">238</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23438/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.html">233</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23438/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount_base.html">292</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23438/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount.html">306</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>MakeStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23438/perf-test-react-components/MakeStyles-mount_base.html">1599</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23438/perf-test-react-components/MakeStyles-mount.html">1628</a></td>
            <td>50000</td>
            <td></td>
           </tr></table></p></details>

<div id="perfComment9425"/>

File: packages/react-components/react-components/src/Concepts/Icons/Icon.stories.mdx [hunk 1]
```diff
@@ -1,10 +1,13 @@
 import { Meta } from '@storybook/addon-docs';
+import { Spinner } from '@fluentui/react-spinner';
 
 <Meta title="Concepts/Developer/Icons" />
 
 ## Icons
 
-The Fluent UI SVG icon set is released under the MIT License.
+The Fluent UI SVG icon set is released under the MIT License. There are three ways you can consume the fluent icon set in your application, with perks and drawbacks to each.
```
Comment (spmonahan) [inline packages/react-components/react-components/src/Concepts/Icons/Icon.stories.mdx:8]:
```suggestion
The Fluent UI icon set is released under the MIT License. There are three ways you can consume the Fluent icon set in your application, with perks and drawbacks to each.
```

The icons are not always consumed as SVGs.

File: packages/react-components/react-components/src/Concepts/Icons/Icon.stories.mdx [hunk 2]
```diff
@@ -25,8 +28,122 @@ import { AccessTime24Filled } from '@fluentui/react-icons';
 
 Each icon also accepts `className` and `primaryFill` props for styling.
 
+### Fluent Icons as Fonts
+
+The fluent icons also can be consumed as a font. The API is the same as the react-element icons, and there are tools to allow you to more seamlessly integrate the font icons
```
Comment (spmonahan) [inline packages/react-components/react-components/src/Concepts/Icons/Icon.stories.mdx:33]:
```suggestion
Fluent icons also can be consumed as a font. The API is the same as the React element icons, and there are tools to allow you to more seamlessly integrate the font icons
```

File: packages/react-components/react-components/src/Concepts/Icons/Icon.stories.mdx [hunk 3]
```diff
@@ -25,8 +28,122 @@ import { AccessTime24Filled } from '@fluentui/react-icons';
...
+  plugins: [
+    // Include this plugin
+    new FluentUIReactIconsFontSubsettingPlugin(),
+  ],
+};
+```
+
+### Fluent Icons as SVG Sprites
+
+Another way that you can consume the fluent icons is by using them as svg-sprites. The fluent svg-sprites are located in the package `@fluentui/svg-sprites`.
```
Comment (spmonahan) [inline packages/react-components/react-components/src/Concepts/Icons/Icon.stories.mdx:68]:
```suggestion
Fluent icons can be consumed as SVG sprites. The Fluent SVG sprites are located in the package `@fluentui/svg-sprites`.
```

File: packages/react-components/react-components/src/Concepts/Icons/Icon.stories.mdx [hunk 4]
```diff
@@ -25,8 +28,122 @@ import { AccessTime24Filled } from '@fluentui/react-icons';
...
+
+## Pros and Cons of Each Icon Approach
+
+#### Icons as React Elements
+
+Pros
+
+- The visual fidelity of the icons imported here are the highest
+- These icons are the easiest to use right out of the gate. They require no setup except for importing the icons that you need from `@fluentui/react-icons`
+- Has treeshaking enabled when using esm node modules
```
Comment (spmonahan) [inline packages/react-components/react-components/src/Concepts/Icons/Icon.stories.mdx:114]:
```suggestion
- Has tree shaking enabled when using ES modules
```

File: packages/react-components/react-components/src/Concepts/Icons/Icon.stories.mdx [hunk 5]
```diff
@@ -25,8 +28,122 @@ import { AccessTime24Filled } from '@fluentui/react-icons';
...
+
+Cons
+
+- If you are not using esm modules in your application, then treeshaking will not work properly, and icons that you don't use in the application
```
Comment (spmonahan) [inline packages/react-components/react-components/src/Concepts/Icons/Icon.stories.mdx:118]:
```suggestion
- If you are not using ES modules in your application, then tree shaking will not work properly, and icons that you don't use in the application
```

File: packages/react-components/react-components/src/Concepts/Icons/Icon.stories.mdx [hunk 6]
```diff
@@ -25,8 +28,122 @@ import { AccessTime24Filled } from '@fluentui/react-icons';
...
+
+Pros
+
+- Render perf is better in applications, as the font files are downloaded on application load and get rendered as an image
+- The font subsetting webpack plugin detects the icons that you use in your application and downloads only those to your application
+
+Cons
+
+- There is a slight dropoff in the visual fidelity of the font icons, but this is not noticeable at most sizes
+- There is more setup required to use the font icons than using the svg based icons i.e, setting up the webpack plugin
```
Comment (spmonahan) [inline packages/react-components/react-components/src/Concepts/Icons/Icon.stories.mdx:131]:
```suggestion
- There is more setup required to use the font icons than using the SVG based icons i.e, setting up the webpack plugin
```

File: packages/react-components/react-components/src/Concepts/Icons/Icon.stories.mdx [hunk 7]
```diff
@@ -25,8 +28,122 @@ import { AccessTime24Filled } from '@fluentui/react-icons';
...
+
+#### Icons as Svg Sprites
```
Comment (spmonahan) [inline packages/react-components/react-components/src/Concepts/Icons/Icon.stories.mdx:133]:
```suggestion
#### Icons as SVG Sprites
```

File: packages/react-components/react-components/src/Concepts/Icons/Icon.stories.mdx [hunk 8]
```diff
@@ -25,8 +28,122 @@ import { AccessTime24Filled } from '@fluentui/react-icons';
 
 Each icon also accepts `className` and `primaryFill` props for styling.
 
+### Fluent Icons as Fonts
+
+Fluent icons also can be consumed as a font. The API is the same as the React element icons, and there are tools to allow you to more seamlessly integrate the font icons
```
Comment (khmakoto) [inline packages/react-components/react-components/src/Concepts/Icons/Icon.stories.mdx:33]: _nit:_

```suggestion
Fluent icons can also be consumed as a font. The API is the same as the React element icons, and there are tools to allow you to more seamlessly integrate the font icons
```

File: packages/react-components/react-components/src/Concepts/Icons/Icon.stories.mdx [hunk 9]
```diff
@@ -25,8 +28,122 @@ import { AccessTime24Filled } from '@fluentui/react-icons';
...
+into your application. You will also be able to only bundle the font icons you need in your application using the the package `@fluentui/react-icons-font-subsetting-webpack-plugin`
```
Comment (khmakoto) [inline packages/react-components/react-components/src/Concepts/Icons/Icon.stories.mdx:34]: _nit:_

```suggestion
into your application. You will also be able to only bundle the font icons you need in your application using the `@fluentui/react-icons-font-subsetting-webpack-plugin` package.
```

File: packages/react-components/react-components/src/Concepts/Icons/Icon.stories.mdx [hunk 10]
```diff
@@ -1,10 +1,13 @@
...
+import { Spinner } from '@fluentui/react-spinner';
 
 <Meta title="Concepts/Developer/Icons" />
 
 ## Icons
 
-The Fluent UI SVG icon set is released under the MIT License.
+The Fluent UI icon set is released under the MIT License. There are three ways you can consume the Fluent icon set in your application, with perks and drawbacks to each.
+
+### Fluent Icons as React Elements
```
Comment (khmakoto) [inline packages/react-components/react-components/src/Concepts/Icons/Icon.stories.mdx:10]: _nit:_

```suggestion
### Fluent icons as React elements
```

File: packages/react-components/react-components/src/Concepts/Icons/Icon.stories.mdx [hunk 11]
```diff
@@ -25,8 +28,122 @@ import { AccessTime24Filled } from '@fluentui/react-icons';
...
+  const assetId = `${icon}_${size}_${variant}`;
+  return (
+    <svg>
+      <use href={`/sprites/${assetId}.sprite.svg#${assetId}`} />
+    </svg>
+  );
+};
+```
+
+## Pros and Cons of Each Icon Approach
```
Comment (khmakoto) [inline packages/react-components/react-components/src/Concepts/Icons/Icon.stories.mdx:106]: _nit:_

```suggestion
## Pros and cons of each approach
```

File: packages/react-components/react-components/src/Concepts/Icons/Icon.stories.mdx [hunk 12]
```diff
@@ -25,8 +28,122 @@ import { AccessTime24Filled } from '@fluentui/react-icons';
...
+
+#### Icons as React Elements
```
Comment (khmakoto) [inline packages/react-components/react-components/src/Concepts/Icons/Icon.stories.mdx:108]: _nit:_

```suggestion
#### Icons as React elements
```

File: packages/react-components/react-components/src/Concepts/Icons/Icon.stories.mdx [hunk 13]
```diff
@@ -25,8 +28,122 @@ import { AccessTime24Filled } from '@fluentui/react-icons';
...
+
+Pros
+
+- The visual fidelity of the icons imported here are the highest
```
Comment (khmakoto) [inline packages/react-components/react-components/src/Concepts/Icons/Icon.stories.mdx:112]: _nit:_

```suggestion
- The visual fidelity of the icons imported here is the highest.
```

File: packages/react-components/react-components/src/Concepts/Icons/Icon.stories.mdx [hunk 14]
```diff
@@ -25,8 +28,122 @@ import { AccessTime24Filled } from '@fluentui/react-icons';
...
+- These icons are the easiest to use right out of the gate. They require no setup except for importing the icons that you need from `@fluentui/react-icons`
+- Has tree shaking enabled when using ES modules
```
Comment (khmakoto) [inline packages/react-components/react-components/src/Concepts/Icons/Icon.stories.mdx:114]: _nit:_

```suggestion
- These icons are the easiest to use right out of the gate. They require no setup except for importing the icons that you need from `@fluentui/react-icons`.
- Has tree shaking enabled when using ES modules.
```

File: packages/react-components/react-components/src/Concepts/Icons/Icon.stories.mdx [hunk 15]
```diff
@@ -25,8 +28,122 @@ import { AccessTime24Filled } from '@fluentui/react-icons';
...
+
+Cons
+
+- If you are not using ES modules in your application, then tree shaking will not work properly, and icons that you don't use in the application
+  could get bundled in along with the icons you do use. This means your perf and bundle size would take a hit importing the icons.
+
+#### Icons as Font
```
Comment (khmakoto) [inline packages/react-components/react-components/src/Concepts/Icons/Icon.stories.mdx:121]: _nit:_

```suggestion
#### Icons as fonts
```

File: packages/react-components/react-components/src/Concepts/Icons/Icon.stories.mdx [hunk 16]
```diff
@@ -25,8 +28,122 @@ import { AccessTime24Filled } from '@fluentui/react-icons';
...
+
+Pros
+
+- Render perf is better in applications, as the font files are downloaded on application load and get rendered as an image
+- The font subsetting webpack plugin detects the icons that you use in your application and downloads only those to your application
```
Comment (khmakoto) [inline packages/react-components/react-components/src/Concepts/Icons/Icon.stories.mdx:126]: _nit:_

```suggestion
- Render perf is better in applications, as the font files are downloaded on application load and get rendered as an image.
- The font subsetting webpack plugin detects the icons that you use in your application and only downloads those to your application.
```

File: packages/react-components/react-components/src/Concepts/Icons/Icon.stories.mdx [hunk 17]
```diff
@@ -25,8 +28,122 @@ import { AccessTime24Filled } from '@fluentui/react-icons';
...
+
+Cons
+
+- There is a slight dropoff in the visual fidelity of the font icons, but this is not noticeable at most sizes
+- There is more setup required to use the font icons than using the SVG based icons i.e, setting up the webpack plugin
```
Comment (khmakoto) [inline packages/react-components/react-components/src/Concepts/Icons/Icon.stories.mdx:131]: _nit:_

```suggestion
- There is a slight dropoff in the visual fidelity of the font icons, but this is not noticeable at most sizes.
- There is more setup required to use the font icons than using the SVG based icons i.e, setting up the webpack plugin.
```

File: packages/react-components/react-components/src/Concepts/Icons/Icon.stories.mdx (hunk 7 above)
Comment (khmakoto) [reply packages/react-components/react-components/src/Concepts/Icons/Icon.stories.mdx:133]:
```suggestion
#### Icons as SVG sprites
```

File: packages/react-components/react-components/src/Concepts/Icons/Icon.stories.mdx [hunk 18]
```diff
@@ -25,8 +28,122 @@ import { AccessTime24Filled } from '@fluentui/react-icons';
...
+
+#### Icons as Svg Sprites
+
+Pros
+
+- You can request and cache the icons at runtime. This drastically reduces the number of HTTP requests, improving render perf right out of the gate
+- You can use a simple `svg` html tag with the `use` tag to quickly and efficiently insert the icons you want to use in your application
```
Comment (khmakoto) [inline packages/react-components/react-components/src/Concepts/Icons/Icon.stories.mdx:138]:
```suggestion
- You can request and cache the icons at runtime. This drastically reduces the number of HTTP requests, improving render perf right out of the gate.
- You can use a simple `svg` html tag with the `use` tag to quickly and efficiently insert the icons you want to use in your application.
```

File: packages/react-components/react-components/src/Concepts/Icons/Icon.stories.mdx [hunk 19]
```diff
@@ -25,8 +28,122 @@ import { AccessTime24Filled } from '@fluentui/react-icons';
...
+
+Cons
+
+- Using symbol sprites won't work for multicolored icons
+- Any new icons used has to be cached again
```
Comment (khmakoto) [inline packages/react-components/react-components/src/Concepts/Icons/Icon.stories.mdx:143]: _nit:_

```suggestion
- Using symbol sprites won't work for multicolored icons.
- Any new icons used have to be cached again.
```

**Review by khmakoto — APPROVED** [review]

**Pull request merged**